api/v1/services/RegistrationService.js

```diff
@@ -109,6 +113,99 @@ function _adjustRelatedObjects(model, adjustments, t) {
 }
 
 /**
+ * Adds people to the correct mailing lists based on decisions
+ * @param  {Object} oldAttributes	the old set of attributes for a user
+ * @param  {Object} newAttributes	the new set of attributes for a user
+ * @return {Promise<MailingListUser>}	a promise with the save result 
+ */
+function _addToMailingList(oldAttributes, newAttributes){
+	//Status not finalized or nothing has changed, don't add to any list
+	if(_.isUndefined(newAttributes.status) || newAttributes.status === "PENDING"){
+		return;
+	}
+
+	var oldWave = oldAttributes.wave;
+	var newWave = newAttributes.wave;
+	var oldStatus = oldAttributes.status;
+	var newStatus = newAttributes.status;
+	var curUser = {}
+
+	//If the status of the user has just been finalized - this is the initial decision
+	if(oldAttributes.status === "PENDING" && newAttributes.status !== "PENDING"){
+		return User.findById(oldAttributes.userId)
+			.then(function (user){
+				curUser = user.attributes;
+				if(newAttributes.status == "ACCEPTED"){
+					listName = "wave_" + newAttributes.wave;
+				}else if(newAttributes.status == "REJECTED"){
+					listName = "rejected";
+				}else{
+					listName = "waitlisted";
+				}
+				return MailingList.findByName(listName);
+			})
+			.then(function (newList) {
+				return MailService.addToList(curUser, newList.attributes);
+			});
+	}
+	//Applicant's wave was changed
+	else if(oldWave != newWave && oldStatus === newStatus && newStatus === "ACCEPTED"){
+		var oldListName = "wave_" + oldWave;
+		var newListName = "wave_" + newWave;
+
+		return User.findById(oldAttributes.userId)
+			.then(function (user) {
+				curUser = user;
+				return MailingList.findByName(oldListName);
+			})
+			.then(function (oldList) {
+				return MailService.removeFromList(curUser, oldList.attributes);
+			})
+			.then(function () {
+				return MailingList.findByName(newListName);
+			})
+			.then(function (newList) {
+				return MailService.addToList(curUser, newList.attributes);
+			});		
+	}
+	//Applicant accepted off of waitlist
+	else if(oldStatus === "WAITLISTED" && newStatus === "ACCEPTED"){
+		return User.findById(oldAttributes.userId)
+			.then(function (user) {
+				curUser = user;
+				return MailingList.findByName("waitlisted");
+			})
+			.then(function (waitList) {
+				return MailService.removeFromList(curUser, waitList.attributes);
+			})
+			.then(function () {
+				var newListName = "wave_" + newWave;
+				return MailingList.findByName(newListName);
+			})
+			.then(function (newList) {
+				return MailService.addToList(curUser, newList.attributes);
+			});
+	}
+	//Applicant rejected off of waitlist
+	else if(oldStatus === "WAITLISTED" && newStatus === "REJECTED"){
```

Comment:
Is there a check for when someone goes from accepted to waitlist (or rejected)? I feel like this is a very similar situation that would just require taking them off of the wave they're on (if the wave was defined in the old attributes). 

I know it seems like it wouldn't happen, but it's an easy mistake for reviewers to make.